Index: nwchem-4.6/src/geom/geom_hnd.F
===================================================================
RCS file: /msrc/proj/mss/nwchem/src/geom/geom_hnd.F,v
retrieving revision 1.44
retrieving revision 1.44.2.1
diff -u -r1.44 -r1.44.2.1
--- nwchem-4.6/src/geom/geom_hnd.F	28 Oct 2003 19:54:48 -0000	1.44
+++ nwchem-4.6/src/geom/geom_hnd.F	24 Sep 2004 17:47:08 -0000	1.44.2.1
@@ -1,5 +1,5 @@
 *
-* $Id$
+* $Id$
 *
       subroutine geom_bandbi(geom)
       implicit none
@@ -3884,6 +3884,7 @@
       DIMENSION   NZMT(5,MAXGEO)
       DIMENSION   ZVAL(3,MAXGEO)
       DIMENSION   ERRMSG(3)
+      double precision numd
       DATA ERRMSG    /'PROGRAM ','STOP IN ','- ZXYZ -'/
       DATA ZERO,ONE  /0.0D+00,1.0D+00/
       DATA TWO,THREE /2.0D+00,3.0D+00/
@@ -4057,8 +4058,11 @@
                         BET=ZVAL(3,IAT)*PIFAC
                         ALP=PHI
                         GAM  =ACOS((RCB**2+RCA**2-RAB**2)/(TWO*RCB*RCA))
-                        THETA=ACOS(( COS(BET)- COS(ALP)* COS(GAM))/
-     1                             (           SIN(ALP)* SIN(GAM))  )
+                     numd=(COS(BET)- COS(ALP)* COS(GAM))/
+     /                    (SIN(ALP)* SIN(GAM))
+                     if(numd.gt.1d0) numd=1d0
+                     if(numd.lt.1d0) numd=-1d0
+                       THETA=ACOS(numd)
                         IF(NZMT(5,IAT).EQ.-1) THEN
                            THETA=-THETA
                         ENDIF
