--- Makefile.arm64.orig	2023-09-03 13:58:32.000000000 -0700
+++ Makefile.arm64	2023-11-06 14:34:50.055745712 -0800
@@ -8,7 +8,7 @@
 endif
 ifneq (1, $(filter 1,$(GCCVERSIONGT4) $(ISCLANG)))
 CCOMMON_OPT += -march=armv8-a
-ifneq ($(F_COMPILER), NAG)
+ifneq ($(F_COMPILER),$(findstring $(F_COMPILER),PGI NAG))
 FCOMMON_OPT += -march=armv8-a
 endif
 
@@ -18,7 +18,7 @@
 
 ifeq ($(CORE), ARMV8)
 CCOMMON_OPT += -march=armv8-a
-ifneq ($(F_COMPILER), NAG)
+ifneq ($(F_COMPILER),$(findstring $(F_COMPILER),PGI NAG))
 FCOMMON_OPT += -march=armv8-a
 endif
 endif
@@ -71,18 +71,18 @@
 ifeq (1, $(filter 1,$(GCCVERSIONGTEQ7) $(ISCLANG)))
 ifeq (1, $(filter 1,$(GCCVERSIONGTEQ9) $(ISCLANG)))
 CCOMMON_OPT += -march=armv8.2-a -mtune=neoverse-n1
-ifneq ($(F_COMPILER), NAG)
+ifneq ($(F_COMPILER),$(findstring $(F_COMPILER),PGI NAG))
 FCOMMON_OPT += -march=armv8.2-a -mtune=neoverse-n1
 endif
 else
 CCOMMON_OPT += -march=armv8.2-a -mtune=cortex-a72
-ifneq ($(F_COMPILER), NAG)
+ifneq ($(F_COMPILER),$(findstring $(F_COMPILER),PGI NAG))
 FCOMMON_OPT += -march=armv8.2-a -mtune=cortex-a72
 endif
 endif
 else
 CCOMMON_OPT += -march=armv8-a -mtune=cortex-a72
-ifneq ($(F_COMPILER), NAG)
+ifneq ($(F_COMPILER),$(findstring $(F_COMPILER),PGI NAG))
 FCOMMON_OPT += -march=armv8-a -mtune=cortex-a72
 endif
 endif
@@ -162,18 +162,18 @@
 ifeq (1, $(filter 1,$(GCCVERSIONGTEQ7) $(ISCLANG)))
 ifeq (1, $(filter 1,$(GCCVERSIONGTEQ8) $(ISCLANG)))
 CCOMMON_OPT += -march=armv8.2-a -mtune=cortex-a55
-ifneq ($(F_COMPILER), NAG)
+ifneq ($(F_COMPILER),$(findstring $(F_COMPILER),PGI NAG))
 FCOMMON_OPT += -march=armv8.2-a -mtune=cortex-a55
 endif
 else
 CCOMMON_OPT += -march=armv8.2-a -mtune=cortex-a53
-ifneq ($(F_COMPILER), NAG)
+ifneq ($(F_COMPILER),$(findstring $(F_COMPILER),PGI NAG))
 FCOMMON_OPT += -march=armv8.2-a -mtune=cortex-a53
 endif
 endif
 else
 CCOMMON_OPT += -march=armv8-a -mtune=cortex-a53
-ifneq ($(F_COMPILER), NAG)
+ifneq ($(F_COMPILER),$(findstring $(F_COMPILER),PGI NAG))
 FCOMMON_OPT += -march=armv8-a -mtune=cortex-a53
 endif
 endif
@@ -181,21 +181,21 @@
 
 ifeq ($(CORE), THUNDERX)
 CCOMMON_OPT += -march=armv8-a -mtune=thunderx
-ifneq ($(F_COMPILER), NAG)
+ifneq ($(F_COMPILER),$(findstring $(F_COMPILER),PGI NAG))
 FCOMMON_OPT += -march=armv8-a -mtune=thunderx
 endif
 endif
 
 ifeq ($(CORE), FALKOR)
 CCOMMON_OPT += -march=armv8-a -mtune=falkor
-ifneq ($(F_COMPILER), NAG)
+ifneq ($(F_COMPILER),$(findstring $(F_COMPILER),PGI NAG))
 FCOMMON_OPT += -march=armv8-a -mtune=falkor
 endif
 endif
 
 ifeq ($(CORE), THUNDERX2T99)
 CCOMMON_OPT += -march=armv8.1-a -mtune=thunderx2t99
-ifneq ($(F_COMPILER), NAG)
+ifneq ($(F_COMPILER),$(findstring $(F_COMPILER),PGI NAG))
 FCOMMON_OPT += -march=armv8.1-a -mtune=thunderx2t99
 endif
 endif
@@ -208,12 +208,12 @@
 else
 CCOMMON_OPT += -mtune=thunderx2t99
 endif
-ifneq ($(F_COMPILER), NAG)
+ifneq ($(F_COMPILER), PGI NAG)
 FCOMMON_OPT += -march=armv8.3-a -mtune=thunderx3t110
 endif
 else
 CCOMMON_OPT += -march=armv8.1-a -mtune=thunderx2t99
-ifneq ($(F_COMPILER), NAG)
+ifneq ($(F_COMPILER), PGI NAG)
 FCOMMON_OPT += -march=armv8.1-a -mtune=thunderx2t99
 endif
 endif
@@ -221,7 +221,7 @@
 
 ifeq ($(CORE), VORTEX)
 CCOMMON_OPT += -march=armv8.3-a
-ifneq ($(F_COMPILER), NAG)
+ifneq ($(F_COMPILER),$(findstring $(F_COMPILER),PGI NAG))
 FCOMMON_OPT += -march=armv8.3-a
 endif
 endif
@@ -229,7 +229,7 @@
 ifeq (1, $(filter 1,$(GCCVERSIONGTEQ9) $(ISCLANG)))
 ifeq ($(CORE), TSV110)
 CCOMMON_OPT += -march=armv8.2-a -mtune=tsv110
-ifneq ($(F_COMPILER), NAG)
+ifneq ($(F_COMPILER),$(findstring $(F_COMPILER),PGI NAG))
 FCOMMON_OPT += -march=armv8.2-a -mtune=tsv110
 endif
 endif
@@ -241,7 +241,7 @@
 ifeq  ($(ISCLANG), 0)
 CCOMMON_OPT += -mtune=emag
 endif
-ifneq ($(F_COMPILER), NAG)
+ifneq ($(F_COMPILER), PGI NAG)
 FCOMMON_OPT += -march=armv8-a -mtune=emag
 endif
 endif
