--- Makefile.system	2020-12-12 09:19:29.000000000 -0800
+++ Makefile.system	2021-01-19 17:00:32.000000000 -0800
@@ -586,8 +586,12 @@
 endif
 
 ifeq ($(C_COMPILER), CLANG)
+ifeq ($(shell uname -s),Darwin)
+CCOMMON_OPT    += -Xpreprocessor -fopenmp
+else
 CCOMMON_OPT    += -fopenmp
 endif
+endif
 
 ifeq ($(C_COMPILER), INTEL)
 CCOMMON_OPT    += -fopenmp
@@ -813,6 +809,13 @@
 ifeq ($(ARCH), riscv64)
 NO_BINARY_MODE  = 1
 BINARY_DEFINED  = 1
+ifdef INTERFACE64
+ifneq ($(INTERFACE64), 0)
+ifeq ($(F_COMPILER), GFORTRAN)
+FCOMMON_OPT +=  -fdefault-integer-8
+endif
+endif
+endif
 endif
 
 ifeq ($(ARCH), loongarch64)
