--- CMakeLists.txt.org	2021-06-25 19:06:34.000000000 -0700
+++ CMakeLists.txt	2021-06-25 19:06:53.000000000 -0700
@@ -12,7 +12,7 @@
 
 if (UNIX)
    if ("${CMAKE_Fortran_COMPILER_ID}" STREQUAL "Intel")
-      set( CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fltconsistency -fp_port" )
+      set( CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fltconsistency" )
    endif ()
 endif ()
 
@@ -77,7 +77,7 @@
 
 if (UNIX)
    if ("${CMAKE_Fortran_COMPILER_ID}" STREQUAL "Intel")
-      set( CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fltconsistency -fp_port" )
+      set( CMAKE_Fortran_FLAGS "${CMAKE_Fortran_FLAGS} -fltconsistency" )
    endif ()
 endif ()
 
