Index: dft_rdinput.F
===================================================================
RCS file: /msrc/proj/mss/nwchem/src/nwdft/input_dft/dft_rdinput.F,v
retrieving revision 1.119.2.2
diff -u -r1.119.2.2 dft_rdinput.F
--- dft_rdinput.F	31 Aug 2005 01:54:51 -0000	1.119.2.2
+++ dft_rdinput.F	26 Jan 2006 17:01:49 -0000
@@ -784,6 +784,13 @@
      &      call errquit('rdinput: basis set error:', 86, BASIS_ERR)
          if (.not. rtdb_put(rtdb, 'dft:cdfit', mt_log, 1, .true.))
      &      call errquit('rdinput: failed put_rtdb:', 99, RTDB_ERR)
+c
+c     texas breaks when AO_spherical and CD_cartesian
+c
+         if(bas_is_spherical(AO_bas_han).and.
+     N        (.not.bas_is_spherical(CD_bas_han))) 
+     T        call int_app_set_no_texas(rtdb)
+
       else
          if (.not. bas_destroy(CD_bas_han))
      &      call errquit('rdinput:not able to destroy CD_bas:',86,
