Index: int_giaotv10.F
===================================================================
RCS file: /msrc/proj/mss/nwchem/src/NWints/api/int_giaotv10.F,v
retrieving revision 1.1
retrieving revision 1.1.2.1
diff -u -r1.1 -r1.1.2.1
--- int_giaotv10.F	21 Jul 2004 16:13:46 -0000	1.1
+++ int_giaotv10.F	19 Jul 2005 17:46:52 -0000	1.1.2.1
@@ -142,8 +142,8 @@
       if (itype.eq.-1) igen = 1
       if (jtype.eq.-1) jgen = 1
       call spcart_2cBtran(tv10,scr,lscr,
-     &    i_nbf_x,int_nbf_s(itype),itype,igen,trani,
      &    j_nbf_x,int_nbf_s(jtype),jtype,jgen,tranj,
+     &    i_nbf_x,int_nbf_s(itype),itype,igen,trani,
      &    3,.false.)
 c
 c     We now have the integrals in array (nsph_ints,3)
